lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -813,22 +817,71 @@ getByronWallet ctx wid =
     fst <$> getWalletWithCreationTime mkApiByronWallet ctx wid
 
 getByronWalletMigrationInfo
-    :: ctx
+    :: forall ctx s t k.
+       ( DefineTx t
+       , s ~ RndState t
+       , ctx ~ ApiLayer s t k )
+    => ApiLayer s t k
     -> ApiT WalletId
     -> Handler ApiByronWalletMigrationInfo
-getByronWalletMigrationInfo _ _ = throwError err501
+getByronWalletMigrationInfo ctx (ApiT wid) = do
+    (cost,_) <- liftHandler $ withWorkerCtx ctx wid throwE $ \wrk ->
+        W.computeWalletMigrationCostCoinSels wrk wid
+    return $ ApiByronWalletMigrationInfo (Quantity $ fromIntegral cost)
 
 migrateByronWallet
-    :: rndCtx
-    -> seqCtx
+    :: forall t.
+       ( DefineTx t
+       , KeyToAddress t RndKey
+       , KeyToAddress t SeqKey
+       )
+    => ApiLayer (RndState t) t RndKey
+    -> ApiLayer (SeqState t) t SeqKey
     -> ApiT WalletId
        -- ^ Source wallet (Byron)
     -> ApiT WalletId
        -- ^ Target wallet (new-style)
     -> ApiMigrateByronWalletData
     -> Handler [ApiTransaction t]
-migrateByronWallet _rndCtx _seqCtx _sourceWid _targetWid _migrateData =
-    throwError err501
+migrateByronWallet rndCtx seqCtx (ApiT sWid) (ApiT dWid) migrateData = do
+    -- (a) checking if source and destination wallets are in sequential and random
+    -- scheme, respectively --ie., incorrect schemes
+    {--
+    _ <- liftHandler $ withWorkerCtx seqCtx sWid (\_ -> lift $ pure ()) $ \wrk ->
+        W.readWallet wrk sWid
+    when (isRight res) $ liftHandler $ throwE ErrMigrationWalletSeqWalletInSource
+    --}
+
+    -- (b) checking if wallets are present
+    _ <- liftHandler $ withWorkerCtx rndCtx sWid liftE1 $ \wrk ->
+        W.readWallet wrk sWid
+    _ <- liftHandler $ withWorkerCtx seqCtx dWid liftE1 $ \wrk ->
+        W.readWallet wrk dWid
+
+    -- (c) get coin selections
+    (_,coinSels) <- liftHandler $ withWorkerCtx rndCtx sWid throwE $ \wrk ->
+        W.computeWalletMigrationCostCoinSels wrk sWid
+
+    --(d) posting txs
+    let (ApiMigrateByronWalletData (ApiT pwd)) = migrateData
+    forM coinSels $ \selection -> do
+        (tx, meta, wit) <- liftHandler $ withWorkerCtx rndCtx sWid liftE2 $ \wrk ->
+            W.signTx wrk sWid pwd selection
```

Comment:
I actually didn't even realize it'd play out so nicely with our already built-in functions :o ..